src/utils/xcode.rs

```diff
@@ -115,12 +122,17 @@ impl XcodeProjectInfo {
             .arg("-project")
             .arg(path.as_ref().as_os_str())
             .output()?;
-        let mut rv: Output = serde_json::from_slice(&p.stdout).unwrap_or_else(|_| {
-            let err_msg = format!("Command `xcodebuild -list -json -project {}` failed to produce a valid JSON output.", path.as_ref().display());
-            panic!("{}", err_msg);
-        });
-        rv.project.path = path.as_ref().canonicalize()?;
-        Ok(rv.project)
+
+        match serde_json::from_slice::<Output>(&p.stdout) {
+            Ok(mut rv) => {
+                rv.project.path = path.as_ref().canonicalize()?;
+                Ok(rv.project)
+            }
+            Err(e) => {
+                warn!("Command `xcodebuild -list -json -project {}` failed to produce a valid JSON output. Your .xcodeproj might be malformed.", path.as_ref().display());
```

Comment:
I would start with what is wrong and then go more into detail.
```suggestion
                warn!("Your .xcodeproj might be malformed. Command `xcodebuild -list -json -project {}` failed to produce a valid JSON output.", path.as_ref().display());
```